contracts/SpokePool.sol

```diff
@@ -1339,12 +1339,7 @@ abstract contract SpokePool is
         // fillDeadline is relative to the destination chain.
         // Don’t allow fillDeadline to be more than several bundles into the future.
         // This limits the maximum required lookback for dataworker and relayer instances.
-        // Also, don't allow fillDeadline to be in the past. This poses a potential UX issue if the destination
-        // chain time keeping and this chain's time keeping are out of sync but is not really a practical hurdle
-        // unless they are significantly out of sync or the depositor is setting very short fill deadlines. This latter
-        // situation won't be a problem for honest users.
-        if (params.fillDeadline < currentTime || params.fillDeadline > currentTime + fillDeadlineBuffer)
-            revert InvalidFillDeadline();
+        if (params.fillDeadline > currentTime + fillDeadlineBuffer) revert InvalidFillDeadline();
```

Comment:
Is there anywhere in the offchain stack that would break if fill deadline was 0?

Reply:
The BundleDataClient already handles fillDeadline 0 OK and explicit notehat it's intentional: https://github.com/across-protocol/sdk/blob/db244fbaec15dc049452cd104ca7bec5856ad3da/src/clients/BundleDataClient/BundleDataClient.ts#L815

There are a couple of places in the BundleDataClient where we'll need updates to handle pre-fills, but it's mostly tangential to this SpokePool change. I left comments on what we need to verify in this PR: https://github.com/across-protocol/sdk/pull/834